packages/base-zone/tools/testers.js

```diff
@@ -0,0 +1,120 @@
+import * as g from './greeter.js';
+
+/**
+ * @param {import('ava').Assertions} t
+ * @param {import('../src/index.js').Zone} rootZone
+ */
+export const testMakeOnce = (t, rootZone) => {
+  const subZone = rootZone.subZone('sub');
+  const a = subZone.makeOnce('a', () => 'A');
+  t.is(a, 'A');
+  t.throws(() => subZone.makeOnce('a', () => 'A'), {
+    message: /has already been used/,
+  });
+  const nonPassable = harden({
+    hello() {
+      return 'world';
+    },
+  });
+  t.is(rootZone.isStorable(nonPassable), false);
+  t.is(subZone.isStorable(123), true);
+  t.throws(() => rootZone.makeOnce('nonPassable', () => nonPassable), {
+    message: /is not storable/,
+  });
+};
+
+// CAUTION: Do not modify this list; it exists to ensure that future versions
+// of @agoric/zone are compatible with the baggage created by older versions,
```

Comment:
Mixing concerns. This won't go to Endo/zone

Reply:
Actually, it will for historical reasons.  Without this list, the heap zone won't reveal problems that would surface with durable zones, which is one of the main reasons to use zones in the first place.

Until we have an upgrade strategy to move to a different key name allocation policy, this "mixed concern" needs to be enforced.

Reply:
Ah. When it moves I expect we'll reword it to emphasize Endo's independence.